src/libraries/Microsoft.Extensions.DependencyModel/src/Microsoft.Extensions.DependencyModel.csproj

```diff
@@ -37,7 +37,9 @@ By default, the dependency manifest contains information about the application's
 
   <ItemGroup Condition="'$(TargetFrameworkIdentifier)' != '.NETCoreApp'">
     <PackageReference Include="System.Buffers" Version="$(SystemBuffersVersion)" />
-    <PackageReference Include="System.Memory" Version="$(SystemMemoryVersion)" />
+    <!-- 4.6.1 is a temporary workaround to prevent failure coming from System.Runtime.CompilerServices.Unsafe
+         wrong assembly version - https://github.com/dotnet/maintenance-packages/pull/221 -->
+    <PackageReference Include="System.Memory" Version="4.6.1" />
```

Comment:
You'll need every package that might bring this in to do the same -- which is why I was suggesting to switch to packageReference for System.Text.Encodings.Web and  System.Text.Json and use `ToolSetVersion` for them.

Essentially we should have a `*ToolsetVersion` set of properties for every package that's in MSBuild that we might want to use, and if we have a library that we want to run in that context all of its dependencies need to be on those versions.

Reply:
I changed System.Buffers and System.Memory to the toolset versions, and then also changed System.Text.Encodings.Web and System.Text.Json to PackageReferences that use the toolset version as well.

Now I get an error when building and packing saying that both Web and Json are likely unnecessary and won't be pruned. If I comment those two package references, Web and json, I am able to build and pack successfully.

Do we still need these two package references anyway? If yes, what should I do to resolve the error?

Reply:
Looks like I need to fine tune the tfms.

Reply:
You only reference those packages when they arent part of the framework.

Reply:
I added a condition that I think should work and be maintainable.